src/grpc/retry-interceptor.ts

```diff
@@ -92,48 +62,60 @@ export class RetryInterceptor {
-                        );
-                        savedMessageNext(savedReceiveMessage);
-                        next(status);
-                      }
-                    } else {
+                    const whenToRetry =
+                      retryStrategy.determineWhenToRetryRequest({
+                        grpcStatus: status,
+                        grpcRequest: options.method_definition,
+                        attemptNumber: attempts,
+                      });
+
+                    if (whenToRetry === null) {
+                      logger.debug(
+                        `Request not eligible for retry: path: ${options.method_definition.path}; retryable status code: ${status.code}; number of attempts (${attempts}).`
+                      );
                       savedMessageNext(savedReceiveMessage);
-                      next({code: status.code});
+                      next(status);
+                    } else {
+                      `Request eligible for retry: path: ${options.method_definition.path}; response status code: ${status.code}; number of attempts (${attempts}); will retry in ${whenToRetry}ms`;
+                      setTimeout(() => retry(message, metadata), whenToRetry);
                     }
                   },
                 });
                 newCall.sendMessage(savedSendMessage);
                 newCall.halfClose();
               };
-              if (retryableGrpcStatusCodes.includes(status.code)) {
-                logger.debug(
-                  `Request path: ${options.method_definition.path}; response status code: ${status.code}; number of retries (${retries}) is less than max (${maxRetry}), retrying.`
-                );
-                retry(savedSendMessage, savedMetadata);
-              } else {
-                logger.trace(
-                  `Request path: ${options.method_definition.path}; response status code: ${status.code}; not eligible for retry.`
-                );
+
+              if (status.code === Status.OK) {
                 savedMessageNext(savedReceiveMessage);
                 next(status);
+              } else {
+                const whenToRetry = retryStrategy.determineWhenToRetryRequest({
+                  grpcStatus: status,
+                  grpcRequest: options.method_definition,
+                  attemptNumber: attempts,
+                });
+                if (whenToRetry === null) {
+                  logger.trace(
+                    `Request not eligible for retry: path: ${options.method_definition.path}; response status code: ${status.code}.`
+                  );
+                  savedMessageNext(savedReceiveMessage);
+                  next(status);
+                } else {
+                  `Request eligible for retry: path: ${options.method_definition.path}; response status code: ${status.code}; number of attempts (${attempts}); will retry in ${whenToRetry}ms`;
```

Comment:
Same here.

Reply:
fixed in https://github.com/momentohq/client-sdk-javascript/pull/205/commits/15e2d8ea91e9977d19ba25e87b865123dd5c8b28